tests/bson/bson-iterator-iterator-001.phpt

```diff
@@ -0,0 +1,51 @@
+--TEST--
+MongoDB\BSON\Iterator iterating objects and arrays
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/basic.inc';
+
+$tests = [
+    '{}',
+    '{ "foo": "bar" }',
+    '{ "foo": "bar", "bar": "foo" }',
+    '[]',
+    '["foo", "bar"]',
```

Comment:
Should we use some more complex types here? For instance, iterating a root document with a field containing an embedded document would include a Document instance in the dumped output.